ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt

```diff
@@ -407,4 +417,39 @@ class MainService : Service() {
             offloadingServiceRepositoryImpl.deleteOffloadingService(id)
         }
     }
+
+    private suspend fun findService(name: String): OffloadingService? {
+        val models = modelsRepository.getAllModelsStream()
+        // todo: Improve search methods
+        val model = models.filter { list ->
+            list.any {
+                it.name.contains(name)
+            }
+        }.firstOrNull()?.get(0)
+
+        val modelId = model?.uid
+
+        val services = offloadingServiceRepositoryImpl.getAllOffloadingService()
+        val service = services.filter { list ->
+            list.any {
+                it.modelId == modelId
+            }
+        }.firstOrNull()?.get(0)
+
+        return service
+    }
+
+    private fun runLLM(input: String, messenger: Messenger?) {
+        CoroutineScope(Dispatchers.IO).launch {
+            val service = findService("llama")
+
+            service?.let {
+                startService(it.serviceId)
+                // todo: Support other models
+                runLlama2(input, getIpAddress(isRunningOnEmulator), it.port, NewDataCb(messenger))
+            }?.run {
+                Log.e(TAG, "Not supported LLM")
+            }
```

Comment:
When is this block invoked?

Reply:
I just want to check if there are no LLM related service.

Reply:
https://tourspace.tistory.com/208

I was making mistake when using 'let' incorrectly.
```
1. When not to use LET
1-1. 불변 변수의 null check (When just checking null for an immutable variable, don't use LET)
```